### PR TITLE
fix: fix several bugs related to remake and symbolic indexing

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

File: test/remake.jl [hunk 1]
```diff
@@ -0,0 +1,118 @@
+using ModelingToolkit, SymbolicIndexingInterface
```
Comment (isaacsas) [inline test/remake.jl:1]: Can you add JumpSystems to JumpProblems into this test to? (Which requires JumpProcesses?)

Note that JumpProblem is defined in JumpProcesses and has some remake related dispatches there that might need to be updated (basically they just forward all updates to the wrapped prob, i.e. `jump_problem.prob` which is usually a discrete or ODE problem).

Comment (AayushSabharwal) [reply test/remake.jl:1]: Resolved in https://github.com/SciML/SciMLBase.jl/pull/583/commits/d4d5466a90de0f586567e4d8aaa61c5518f9e4d1

**Comment by ChrisRackauckas** [comment]
Add some inference tests as well.

**Comment by ChrisRackauckas** [comment]
What's next here?

**Comment by AayushSabharwal** [comment]
Nothing, it just needs [this](https://github.com/SciML/ModelingToolkit.jl/pull/2409) to be tagged first, which I assume will happen with MTKv9

**Comment by ChrisRackauckas** [comment]
Looks like there's a circle where tests can't pass?

**Comment by AayushSabharwal** [comment]
MTK tests should pass regardless of this PR

**Comment by ChrisRackauckas** [comment]
What's the status of this one?

**Comment by AayushSabharwal** [comment]
This PR needs https://github.com/SciML/ModelingToolkit.jl/pull/2409 to be tagged

**Comment by AayushSabharwal** [comment]
TODO: Symbolic remake won't work with `SymbolCache` as the system as user-provided symbols. I think now that SII exists, `process_p_u0_symbolic` should be able to be implemented in SciMLBase and be independent of where the symbols come from?

**Comment by AayushSabharwal** [comment]
CI won't pass here until https://github.com/SciML/ModelingToolkit.jl/pull/2409 is merged and tagged. I also have plans to remove `process_p_u0_symbolic` from MTK and just handle it in SciMLBase using SII. This would allow for symbolic initialization using DSLs (`SymbolCache`) as well.

**Comment by isaacsas** [comment]
Is there any update on this or the PRs this requires before it can be merged?

**Pull request closed without merging**